### ISSUE TITLE
bug: [bentoml-cli] `build` failed: Distro is required, got None instead.

### ISSUE THREAD

**Issue by sujan-ml** [bentoml/BentoML#3325]
### Describe the bug

when `bentoml build`, this error occurred. Today I moved to BentoML==1.0.11
it was fine with BentoML==1.0.10 when tried last.

### To reproduce

bentoml build

### Environment

bentoml: 1.0.11
python 3.8

**Comment by mqk** [comment]
For more context, this is presumably occurring because we're using a custom docker base image. Our custom base image start with `FROM python:3.8-slim`. Do we need to set an envvar or something to let BentoML know what distro this is?
`

**Comment by mqk** [comment]
I've tried specifying distro: "debian" in the bentofile.yaml, and that seems to have gotten me past this particular error (but I don't think this is a good long-term solution), but then I run into another issue that seems familiar:
```
bentoml.exceptions.InvalidArgument: Invalid build option: docker.python_version="None", python version must follow standard python semver format, e.g. 3.7.10 
```
This is a result of `docker.python_version` ending up as a string "None" (not `None`), and that causes the version validation to fail. I swear we previously solved for this issue, but I can't find where or how.

**Comment by mqk** [comment]
Found where that `python_version` issue was previously fixed: https://github.com/bentoml/BentoML/pull/2623/files

**Comment by mqk** [comment]
So yeah, this is a regression: https://github.com/bentoml/BentoML/commit/cd2d2ea35db07ac43b8bdfca9e62497c2c49b87d
<img width="489" alt="image" src="https://user-images.githubusercontent.com/1635487/206292942-e93d1173-a596-4e6a-8868-1201cc971d96.png">
By getting rid of those `Optional[str]` you've undone the fix from the PR I linked earlier, and that causes a build problem for users of a custom base image. @aarnphm

**Comment by mqk** [comment]
I've confirmed that when I restore those `t.Optional` then I can get the bento to build. Note that I still have to provide `distro: "debian"` in order to get past the original issue reported here. That's undesirable (imo). Maybe we could use a value of "custom_docker_image" (or similar) for `DockerOptions.distro` when a custom docker image is used, and the we could an a dummy entry in `CONTAINER_METADATA`?

**Comment by aarnphm** [comment]
Hi there, this https://github.com/bentoml/BentoML/pull/3329 should address the regression.

**Comment by mqk** [comment]
Awesome, thanks! So I should close the PR that I submitted, correct?

**Comment by aarnphm** [comment]
yep

**Comment by aarnphm** [comment]
This has been addressed in 1.0.12
